zerver/lib/actions.py

```diff
@@ -2388,16 +2388,22 @@ def gather_subscriptions_helper(user_profile):
     for stream in stream_dicts:
         stream_hash[stream["id"]] = stream
 
+    all_streams = list(get_all_realm_streams(user_profile.realm).values())
```

Comment:
There's no need for this extra query here. You can avoid it by modifying `get_active_streams` above on line 2384 to retrieve all streams (not just streams that the user is subscribed to)